wlcreate/Start-Asian-Love
# Refactor CSS and Image for Home page, Header, and Footer components

## Description

Related Issue: #28 



This PR updates:
* the metadata url for each page
* on the Home page changes img to use next's Image component
* on the Header component changes `useRouter` to `Link`

The PR refactors the CSS in:
* `global.css` variables for colors, fonts, and transition
* `Home.module.css` to use css grid for responsive graphics
* classNames to be more readable
* structure css to have a format of "outside --> in" where the top of the file will be the biggest container and below will be the classes inside of it and media queries at the very bottom
* Home page, Header, and Footer responsive breakpoints



Resources:

* [CSS grid for graphics](https://www.freecodecamp.org/news/how-to-make-your-html-responsive-by-adding-a-single-line-of-css-2a62de81e431/)
* [CSS variables](https://dev.to/samanthaming/css-variables-8di)
* [Header Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)
* [Video reference on NextJS navbar](https://dev.to/keefdrive/create-a-navbar-in-nextjs-using-shared-layouts-and-css-modules-12n)
* [Different cursors](https://www.w3schools.com/cssref/tryit.asp?filename=trycss_cursor)

## Acceptance Criteria



- Reference suggestions from [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)
- Image Optimization (using Nextjs built in [Image component](https://nextjs.org/docs/basic-features/image-optimization))
- Include specific images to render based on the media query (re-size images for what we need)
- Defer offscreen images

## Self-check

Please check all that apply:



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


<img width="1267" alt="Home page with original Header component" src="https://user-images.githubusercontent.com/62153993/119212909-7324a780-ba89-11eb-9042-fbfb10b62ee9.png">
<img width="999" alt="Home page with original graphic rows" src="https://user-images.githubusercontent.com/62153993/119212938-a8c99080-ba89-11eb-82ba-1300d273d6a1.png">
<img width="999" alt="Home page with original Footer component" src="https://user-images.githubusercontent.com/62153993/119212944-b5e67f80-ba89-11eb-9181-19235d2db2ec.png">

### After


<img width="1279" alt="Home page with updated Header component" src="https://user-images.githubusercontent.com/62153993/119212959-d44c7b00-ba89-11eb-936d-ad440c265906.png">
<img width="1279" alt="Home page with updated graphic rows and Footer component" src="https://user-images.githubusercontent.com/62153993/119212967-e4645a80-ba89-11eb-90a4-cf0d20491baf.png">


## Testing Steps / QA Criteria


* switch to this branch `performance-image` or visit the [preview](https://start-asian-love-cqukc4jd7-waverleynyc-gmailcom.vercel.app/)
* test the responsiveness of the app, specifically check for the font size and grid/flexbox changes
